libs/wingsdk/src/std/struct.ts

```diff
@@ -0,0 +1,45 @@
+import { Validator } from "jsonschema";
+import { T1 } from "./generics";
+import { Json } from "./json";
+import { Code, InflightClient } from "../core";
+
+/**
+ * Shared behavior for all structs
+ *
+ * @typeparam T1
+ */
+export class Struct {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): Code {
+    return InflightClient.forType(__filename, this.name);
+  }
+
+  /**
+   * Converts a Json to a Struct
+   *
+   * This macro takes a Json object and the Struct definition file name then calls the validate method
+   *
+   * @macro ((j, s) => {return s._validate(j)})($args$, $self$)
+   */
+  public static fromJson(json: Json): T1 {
+    json;
+    throw new Error("Macro");
+  }
+
+  /** @internal */
+  public static _getValidator(deps?: { [key: string]: any }): Validator {
```

Comment:
Add doc string. What is deps?

Reply:
I was able to refactor the schema to use $defs https://json-schema.org/understanding-json-schema/structuring.html#defs 

rather than adding external schemas as dependencies to the validator. So this now makes the `getSchema()` method return a self contained schema. 🎉 so this method is now gone 

Note:
In a future iteration I would like to get rid of the require() statements within the schema (when referencing parent fields) but that is going to take some heavier refactoring,